Add response metadata in annotated pairs format

timokau opened the pull request:
This adjusts the annotated pairs format (now v2.0) so that each response
is a dictionary instead of a string, now called "response_a" instead of
"text_a". The original text content is stored under the "text" key, with
additional fields for metadata like "model" name.

For consistency with the rename from text_a/text_b to response_a/response_b,
preference indicators are simplified from "text_a"/"text_b" to just "a"/"b".

The goal is to make model identity first-class metadata to make it
easier to compare models independent of preferences.

rdnfn commented on src/inverse_cai/data/annotated_pairs_format.py:
```diff
@@ -20,20 +20,35 @@
+        response_b: A dictionary with at least a 'text' key
+        prompt: An optional prompt string
+    """
+    combined = f"{response_a['text']}|{response_b['text']}|"
```
There is a special case we want to account for: when three models produce the same response to a prompt, then this leads to identical hashes for 3 three different model comparisons - even though they contain different information for the sake of comparing models. This actually happens more frequently than one might expect (e.g. a prompt that just asks for an "ok"). I think we should add the model names into the hash to account for this special case and easily keep track of these different scenarios. If think the efficiency loss is relatively low, as these scenarios are not exactly frequent either.

rdnfn replied:
What do you think?

rdnfn commented:
Thanks a lot for implementing this! See my review comments - further, could you update the README with the new format? Will be great to have it documented there (I already found this very useful for my own dev work).

timokau commented:
I have adjusted the comparison hash computation to include all response information and updated the README to reflect annotated pairs v2.0. Thanks for the feedback!

rdnfn commented:
Thanks @timokau! Is it possible you didn't push a commit yet? I don't think I can see the changes you mention.

timokau commented:
Yes sorry, I forgot to push. Should be all there now.

rdnfn reviewed (approved):
Nice, thanks! Then this all looks very good, will merge.
